src/components/TradeWidget/index.tsx

```diff
@@ -8,9 +8,10 @@ import { tokenListApi } from 'api'
 import TokenRow from './TokenRow'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faExchangeAlt, faPaperPlane } from '@fortawesome/free-solid-svg-icons'
+import { useTokenBalances } from 'hooks/useTokenBalances'
 
-function _getToken(symbol: string, tokens: TokenDetails[]): TokenDetails {
-  return tokens.find(token => token.symbol == symbol)
+function _getToken<T extends TokenDetails>(symbol: string, tokens: T[]): T {
```

Comment:
I would just do:

`function _getToken<T extends TokenDetails>(symbol: string, tokens: T[] = []): T { ... }` (default param)`

instead of the `(token || []).find(...)`

Reply:
Refactored in https://github.com/gnosis/dex-react/pull/163, I'll address this there.